packages/runtime/src/runtimes/react/controls/rich-text.tsx

```diff
@@ -0,0 +1,32 @@
+import dynamic from 'next/dynamic'
+import { ReactNode, useCallback } from 'react'
+import {
+  RichTextControl,
+  RichTextControlData,
+  RichTextControlDefinition,
+} from '../../../controls/rich-text'
+import { forwardNextDynamicRef } from '../../../next'
+import {
+  isPropControllersHandle,
+  PropControllersHandle,
+} from '../../../state/modules/prop-controller-handles'
+
+const Text = forwardNextDynamicRef(patch =>
+  dynamic(() => patch(import('../../../components/builtin/Text'))),
+)
+
+export type RichTextControlValue = ReactNode
+
+export type Descriptors = { text?: RichTextControlDefinition }
+
+export function useRichText(data: RichTextControlData, control: RichTextControl | null) {
+  const refCallback = useCallback(
+    (ref: PropControllersHandle<Descriptors> | HTMLDivElement) => {
```

Comment:
minor nit, the argument provided to the callback ref isn't a ref since it's not mutable and there's no `current` field, it's actually the `current` field itself. Some suggestions for names:

- `refCallback` -> `textCallbackRef`
- `ref` -> `current`
- `ref` -> `handle`